cloud/util/util.go

```diff
@@ -320,10 +320,10 @@ func CreateMachinePoolMachinesIfNotExists(ctx context.Context, params MachinePar
 		}
 		infraMachine := &infrav2exp.OCIMachinePoolMachine{
 			ObjectMeta: metav1.ObjectMeta{
-				Namespace:   params.Namespace,
-				Name:        specMachine.Name,
-				Labels:      labels,
-				Annotations: make(map[string]string),
+				Namespace:    params.Namespace,
+				GenerateName: params.MachinePool.Name,
```

Comment:
Where does the `GenerateName` come from?

Reply:
from the docs 
```
	// GenerateName is an optional prefix, used by the server, to generate a unique
	// name ONLY IF the Name field has not been provided.
	// If this field is used, the name returned to the client will be different
	// than the name passed. This value will also be combined with a unique suffix.
```
It is used to generate a unique name for a k8s object using a prefix